src/core/execute.c

```diff
@@ -2422,6 +2426,7 @@ static int exec_child(
         r = reset_signal_mask();
         if (r < 0) {
                 *exit_status = EXIT_SIGNAL_MASK;
+                *error_message = strdup("Failed to reset signal mask");
```

Comment:
None of these get checked for ENOMEM, but that should be okay -- the caller handles the case when `error_message` is `NULL`.